From stable-bounces@linux.kernel.org Wed Feb 21 14:39:16 2007
From: Josef Whiter <jwhiter@redhat.com>
Date: Wed, 21 Feb 2007 14:37:59 -0800
Subject: gfs2: fix locking mistake
To: mm-commits@vger.kernel.org
Cc: jwhiter@redhat.com, stable@kernel.org, swhiteho@redhat.com
Message-ID: <200702212237.l1LMbxF0028615@shell0.pdx.osdl.net>

From: Josef Whiter <jwhiter@redhat.com>

Fix a locking mistake in the quota code, we do a mutex_lock instead of a
mutex_unlock.

Signed-off-by: Josef Whiter <jwhiter@redhat.com>
Cc: Steven Whitehouse <swhiteho@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/gfs2/quota.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.20.1.orig/fs/gfs2/quota.c
+++ linux-2.6.20.1/fs/gfs2/quota.c
@@ -279,7 +279,7 @@ static int bh_get(struct gfs2_quota_data
 		(bh->b_data + sizeof(struct gfs2_meta_header) +
 		 offset * sizeof(struct gfs2_quota_change));
 
-	mutex_lock(&sdp->sd_quota_mutex);
+	mutex_unlock(&sdp->sd_quota_mutex);
 
 	return 0;
 
